pkg/endpoint/endpoint.go

```diff
@@ -449,6 +450,101 @@ func (e *Endpoint) GetModelRLocked() *models.Endpoint {
 	}
 }
 
+// newHealthz returns the endpoint's healthz object.
+//
+// Must be called with e.Mutex locked.
+func (e *Endpoint) getHealthzModel() *models.EndpointHealthz {
+
+	// figure out build from bpf state & EP state?
+	// figure out policy from?
+	// figure out health from build & policy
+
+	// Duplicated from GetModelRLocked.
+	currentState := models.EndpointState(e.state)
+	if currentState == models.EndpointStateReady && e.Status.CurrentStatus() != OK {
+		currentState = models.EndpointStateNotReady
+	}
+
+	bpfStatus := models.EndpointHealthzStatusOK
+	switch currentState {
+	case models.EndpointStateRegenerating, models.EndpointStateWaitingToRegenerate, models.EndpointStateDisconnecting:
+		bpfStatus = models.EndpointHealthzStatusPending
+
+	case models.EndpointStateCreating:
+		bpfStatus = models.EndpointHealthzStatusBootstrap
+
+	case models.EndpointStateWaitingForIdentity:
+		bpfStatus = models.EndpointHealthzStatusDisabled
+
+	case models.EndpointStateNotReady:
+		bpfStatus = models.EndpointHealthzStatusWarning
+
+	case models.EndpointStateDisconnected:
+		bpfStatus = models.EndpointHealthzStatusDisabled
+	}
+
+	policyStatus := models.EndpointHealthzStatusOK
+	switch currentState {
```

Comment:
Why do you have a duplicate switch? You can assign `policyStatus` and `bpfStatus` at the same time in a single switch

Reply:
Yeah, this was a bad idea. I expected more variation. I collapsed everything into a single switch.